zerver/actions/message_edit.py

```diff
@@ -1296,6 +1268,83 @@ def check_time_limit_for_change_all_propagate_mode(
     )
 
 
+def get_message_edit_request_object(
+    message: Message,
+    user_profile: UserProfile,
+    propagate_mode: str,
+    stream_id: int | None = None,
+    topic_name: str | None = None,
+    content: str | None = None,
+) -> StreamMessageEditRequest | DirectMessageEditRequest:
+    if not message.is_stream_message():
+        # We have already validated the code to have content
+        # as not None.
+        assert content is not None
+        return DirectMessageEditRequest(
+            content=content,
+            orig_content=message.content,
+            is_content_edited=True,
+        )
+
+    is_content_edited = False
+    new_content = message.content
+    if content is not None:
+        is_content_edited = True
+        if content.rstrip() == "":
+            content = "(deleted)"
+        new_content = normalize_body(content)
+
+    is_topic_edited = False
+    topic_resolved = False
+    topic_unresolved = False
+    old_topic_name = message.topic_name()
+    new_topic_name = old_topic_name
```

Comment:
Have kept this so that `topic_name` and `stream` can be set to orginal value if they are not changed and the new value otherwise so that we do not need conditions like `stream = new_stream if new_stream is not None else orig_stream`.
And we have boolean fields for checking which of the things were changed.